src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostBuilderTests.cs

```diff
@@ -665,6 +664,27 @@ public void HostBuilderConfigureDefaultsInterleavesMissingConfigValues()
             Assert.Equal(expectedContentRootPath, env.ContentRootPath);
         }
 
+        [Fact]
+        public void HostBuilderConfigureDefaultsDoesntThrowInDevelopment()
+        {
+            using (var host = new HostBuilder()
+                .ConfigureDefaults(args: null)
+                .ConfigureHostConfiguration(config =>
+                {
+                    config.AddInMemoryCollection(new[]
+                    {
+                        new KeyValuePair<string, string>(HostDefaults.ApplicationKey, "MyProjectReference")
+                    });
+                })
+                .Build())
+            {
+                var env = host.Services.GetRequiredService<IHostEnvironment>();
+
+                Assert.Equal("MyProjectReference", env.ApplicationName);
+                Assert.Equal(Environments.Development, env.EnvironmentName);
```

Comment:
It looks like the test is failing because on Release builds this is `Production`.

```
Assert.Equal() Failure
Expected: Development
Actual:   Production

 at Microsoft.Extensions.Hosting.Tests.HostBuilderTests.HostBuilderConfigureDefaultsDoesntThrowInDevelopment() in /_/src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostBuilderTests.cs:line 684
```